lib/runtime/router.ts

```diff
@@ -251,6 +251,28 @@ export default class Router extends DenaliObject implements RouterDSL {
   }
 
   /**
+   * Returns the URL for a given action. You can supply a params object which
+   * will be used to fill in the dynamic segements of the action's route (if
+   * any).
+   */
+  urlFor(action: string | Action, data: Object): string | null {
+    if (typeof action === 'string') {
+      action = this.container.lookup(`action:${ action }`);
+    }
+    if (!action) {
+      return null;
+    }
+
+    let route: Route;
+    forEach(this.routes, (routes) => {
+      route = find(routes, { action: <Action> action });
+      return !route; // kill the iterator if we found the match
```

Comment:
would that work? find returns the first element that the predicate returns truthy for - meaning because `this.routes` is a hash of arrays indexed by verb (get, post, etc), the logical result of the `find` call would be an array.